sootty/__main__.py

```diff
@@ -163,12 +177,43 @@ def compile(filename, wires, breakpoints, length, start, end, display):
         print(image.source)
 
 
-def reload_query(parser, reload, reload_path):
-    reload = reload_path + reload
-    with open(reload, "r") as rf:
-        cmd = rf.readline()
+def reload_query(
+    parser, filename, wires, breakpoints, length, start, end, display, reload, savefile
+):
+    with open(savefile, "r") as stream:
+        try:
+            dat = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+            parser.print_usage()
+            exit(1)
+        cmd = dat[reload]["query"]
     args = parser.parse_args(shlex.split(cmd))  # using shlex to parse string correctly
 
+    print(args)
+
+    # Updating specifc flags for a relaoded query
+
+    if filename is not None:
+        args.filename = filename
+    if wires is not None:
+        args.wires = wires
+    if breakpoints is not None:
+        args.breakpoints = breakpoints
+    if length is not None:
+        args.length = length
+    if start is not None:
+        args.start = start
+    if end is not None:
+        args.end = end
+
+    print(args)
+
+    if length is not None and end is not None:
+        raise SoottyError(
+            f"Length and end flags should not be provided simultaneously."
+        )
+
     compile(
```

Comment:
This appears to be a reserved name in python3.x, maybe change it to `compile_query`.